database.py

```diff
@@ -62,76 +61,115 @@ def __init__(self, cls_, name, value, description):
     def __reduce__(self):
         """
         Allow unpickling to return the symbol linked to the DeclEnum class.
+
+        :return: method and object reference to unpickle with
+        :rtype: method, (class, attribute)
         """
         return getattr, (self.cls_, self.name)
 
     def __iter__(self):
+        """
+        Provide iterator for the class.
+
+        :return: iterator
+        :rtype: iter
+        """
         return iter([self.value, self.description])
 
     def __repr__(self):
+        """
+        Define object representation when used with display method such as print.
+
+        :return: object representation
+        :rtype: str
+        """
         return "<{name}>".format(name=self.name)
 
 
 class EnumMeta(type):
-    """
-    Generate new DeclEnum classes.
-    """
+    """Generate new DeclEnum classes."""
 
-    def __init__(cls, classname, bases, dict_):
-        cls._reg = reg = cls._reg.copy()
+    def __init__(self, classname, bases, dict_):
+        """Initilize EnumMeta with class, name, value and description."""
+        self._reg = reg = self._reg.copy()
         for k, v in dict_.items():
             if isinstance(v, tuple):
-                sym = reg[v[0]] = EnumSymbol(cls, k, *v)
-                setattr(cls, k, sym)
-        return type.__init__(cls, classname, bases, dict_)
+                sym = reg[v[0]] = EnumSymbol(self, k, *v)
+                setattr(self, k, sym)
+        return type.__init__(self, classname, bases, dict_)
 
-    def __iter__(cls):
-        return iter(cls._reg.values())
+    def __iter__(self):
```

Comment:
Why change from `cls` to `self`?

Reply:
I changed `cls` to `self` since the methods are not class method and hence should have `self` as the first argument.

http://pylint-messages.wikidot.com/messages:e0213